public/upgrade.php

```diff
@@ -1765,3 +1765,8 @@ function upgrade_1842() {
     _db_add_field('mailbox', 'password_expiry', "{DATETIME}"); // when a specific mailbox password expires
     _db_add_field('domain',  'password_expiry', 'int DEFAULT 0'); // expiry applied to mailboxes within that domain
 }
+
+function upgrade_1843() {
+   # Additional field for fetchmail to allow server with non-standard port
+    _db_add_field('fetchmail', 'src_service', "varchar(30) {UTF-8} NOT NULL DEFAULT ''", 'src_server');
```

Comment:
UTF-8 for a port name or number? Really? ;-)

Also, do you have real-world usecases where you need to specify the port, but at the same time can use a port name?

My guess would be that this is a very rare situation, and restricting the port to a port number (integer) would not be annoying to anybody ;-) and at the same time easily avoid that someone can abuse this field to inject something into the generated fetchmail.conf.